webapp/lib/controller.dart

```diff
@@ -1384,28 +1391,52 @@ void setMessageTag(model.Tag tag, model.Message message, model.Conversation conv
   }
 }
 
-void updateMissingTagIds(Set<model.Conversation> conversations, List<model.Tag> tags) {
+void updateMissingTagIds(Set<model.Conversation> conversations, List<model.Tag> tags, List<TagFilterType> filterTypes) {
   var newTagIdsWithMissingInfo = extractTagIdsWithMissingInfo(conversations, tags.toSet());
-  var tagsWithMissingInfo = conversationTags.where((tag) => tag.type == model.NotFoundTagType.NotFound).toSet();
+  var tagsWithMissingInfo = tags.where((tag) => tag.type == model.NotFoundTagType.NotFound).toSet();
   if (newTagIdsWithMissingInfo.isEmpty) {
-    conversationTags.removeWhere((tag) => tag.type == model.NotFoundTagType.NotFound);
+    tags.removeWhere((tag) => tag.type == model.NotFoundTagType.NotFound);
   }
   var tagIdsWithMissingInfo = tagsWithMissingInfo.map((tag) => tag.docId).toSet();
   // remove tags that are no longer missing their info
   var tagIdsToRemove = tagIdsWithMissingInfo.difference(newTagIdsWithMissingInfo);
   if (tagIdsToRemove.isNotEmpty) {
-    var tagsToRemove = tagIdsToTags(tagIdsToRemove, conversationTags);
-    conversationTags.removeWhere((tag) => tagsToRemove.contains(tag));
-    _removeTagsFromFilterMenu({tagsToRemove.first.group: tagsToRemove}, TagFilterType.include);
-    _removeTagsFromFilterMenu({tagsToRemove.first.group: tagsToRemove}, TagFilterType.exclude);
+    var tagsToRemove = tagIdsToTags(tagIdsToRemove, Map.fromEntries(tags.map((t) => MapEntry(t.tagId, t))));
+    tags.removeWhere((tag) => tagsToRemove.contains(tag));
+    var groupsToUpdate = Set();
```

Comment:
You could use `= <String>{}` to provide better type validation.

Reply:
good idea, thanks @danrubel !